arcgisUM.py

```diff
@@ -88,39 +88,84 @@ def getArcGISGroupByTitle(arcGISAdmin, title):
     return None
 
 
-def addCanvasUsersToGroup(instructorLog, group, courseUsers):
-    """Add new users to the ArcGIS group.  """
-    groupNameAndID = util.formatNameAndID(group)
-    
+def addCanvasUsersToGroup(group, courseUsers, instructorLog):
+    """Add new users to a specific ArcGIS group"""
+
     logger.info("addCanvasUsersToGroup: enter")
+    groupNameAndID = util.formatNameAndID(group)
     
     if len(courseUsers) == 0:
         logger.info('No new users to add to ArcGIS Group {}'.format(groupNameAndID))
+        instructorLog += "No new users were added.\n\n"
         return instructorLog
 
     logger.info('Adding Canvas Users to ArcGIS Group {}: {}'.format(groupNameAndID, courseUsers))
     # ArcGIS usernames are U-M uniqnames with the ArcGIS organization name appended.
     arcGISFormatUsers = formatUsersNamesForArcGIS(courseUsers)
     logger.debug("addCanvasUsersToGroup: formatted: {}".format(arcGISFormatUsers))
-    
-    results = group.add_users(arcGISFormatUsers)
-    logger.debug("adding: results: {}".format(results))
 
-    usersNotAdded = results.get('notAdded')
-    """:type usersNotAdded: list"""
     usersCount = len(arcGISFormatUsers)
-    usersCount -= len(usersNotAdded) if usersNotAdded else 0
+    listsOfFormattedUsernames = util.splitListIntoSublists(arcGISFormatUsers, 20)
+    usersNotAdded = []
+
+    for listOfFormattedUsernames in listsOfFormattedUsernames:
+        try:
+            results = group.add_users(listOfFormattedUsernames)
+            logger.debug("adding: results: {}".format(results))
```

Comment:
You should take this opportunity to replace the use of `format()` with f-strings.  At least in the lines of code you touch.  (Changing it all globally would be heavy-handed as well as distract us from the important changes you need us to review now.)

Reply:
I'll do that.

Reply:
Okay, `"{}".format(someVar)` uses in the code I touched now use `f"{someVar}"` strings.